crates/bevy_render/src/render_resource/bind_group.rs

```diff
@@ -42,3 +53,255 @@ impl Deref for BindGroup {
+///     gpu resource, which will be bound as a texture in shaders. The field will be assumed to implement [`Into<Option<Handle<Image>>>`]. In practice,
+///     most fields should be a [`Handle<Image>`](bevy_asset::Handle) or [`Option<Handle<Image>>`]. If the value of an [`Option<Handle<Image>>`] is
+///     [`None`], the [`FallbackImage`] resource will be used instead. This attribute can be used in conjunction with a `sampler` binding attribute
+///    (with a different binding index) if a binding of the sampler for the [`Image`] is also required.
+/// * `sampler(BINDING_INDEX)`
+///     * This field's [`Handle<Image>`](bevy_asset::Handle) will be used to look up the matching [`Sampler`](crate::render_resource::Sampler) gpu
+///     resource, which will be bound as a sampler in shaders. The field will be assumed to implement [`Into<Option<Handle<Image>>>`]. In practice,
+///     most fields should be a [`Handle<Image>`](bevy_asset::Handle) or [`Option<Handle<Image>>`]. If the value of an [`Option<Handle<Image>>`] is
+///     [`None`], the [`FallbackImage`] resource will be used instead. This attribute can be used in conjunction with a `texture` binding attribute
+///     (with a different binding index) if a binding of the texture for the [`Image`] is also required.
+///
+/// Note that fields without field-level binding attributes will be ignored.
+/// ```
+/// # use bevy_render::{color::Color, render_resource::AsBindGroup};
+/// # use bevy_asset::Handle;
+/// #[derive(AsBindGroup)]
+/// struct CoolMaterial {
+///     #[uniform(0)]
+///     color: Color,
+///     this_field_is_ignored: String,
+/// }
+/// ```
+///
+///  As mentioned above, [`Option<Handle<Image>>`] is also supported:
+/// ```
+/// # use bevy_render::{color::Color, render_resource::AsBindGroup, texture::Image};
+/// # use bevy_asset::Handle;
+/// #[derive(AsBindGroup)]
+/// struct CoolMaterial {
+///     #[uniform(0)]
+///     color: Color,
+///     #[texture(1)]
+///     #[sampler(2)]
+///     color_texture: Option<Handle<Image>>,
+/// }
+/// ```
+/// This is useful if you want a texture to be optional. When the value is [`None`], the [`FallbackImage`] will be used for the binding instead, which defaults
+/// to "pure white".
+///
+/// Field uniforms with the same index will be combined into a single binding:
+/// ```
+/// # use bevy_render::{color::Color, render_resource::AsBindGroup};
+/// #[derive(AsBindGroup)]
+/// struct CoolMaterial {
+///     #[uniform(0)]
+///     color: Color,
+///     #[uniform(0)]
+///     roughness: f32,
+/// }
+/// ```
+///
+/// In WGSL shaders, the binding would look like this:
+/// ```wgsl
+/// struct CoolMaterial {
+///     color: vec4<f32>;
+///     roughness: f32;
+/// };
+///
+/// [[group(1), binding(0)]]
+/// var<uniform> material: CoolMaterial;
+/// ```
+///
+/// Some less common scenarios will require "struct-level" attributes. These are the currently supported struct-level attributes:
+/// * `uniform(BINDING_INDEX, ConvertedShaderType)`
```

Comment:
It isn't obvious to me when reading this that `ConvertedShaderType` is a type of a struct that I have implemented separately. I only discovered that when seeing some other examples. I think it is because this looks like a trait name as well rather than something I am supposed to replace.

Reply:
Good call. Ill make this type name a bit more "placeholder-ey".